File: 2/Recursive.java
/**
 * @author Kalle Elmdahl 21/09/07 (updated 21/09/09)
 * The code is used for reversing strings.
 * The code is for an assignment from the KTH-course ID1020
 */
import java.util.Scanner;

/**
 * A class used for printing strings reversed
 */
public class Recursive {
    /**
     * Prints a string in reverse to {@code System.out.print}
     * @param str the string to be reversed
     */
    public static void printCharactersReverse(String s) {
        if(s.isEmpty()) return;
        printCharactersReverse(s.substring(1));
        System.out.print(s.charAt(0));
    }

    /**
     * Main function used for testing the class
     * @param args arguments from program execution
     */
    public static void main(String[] args) {
        System.out.print("\nHello please enter a command: \nrs - reverse a string\nquit - exit the program\n\n");
        Scanner input = new Scanner(System.in);
        outerwhile:
        while(true) {
            System.out.print("$ ");
            String cmd = input.next();

            switch(cmd) {
                case "rs":
                    printCharactersReverse(input.next());
                    break;
                case "quit": break outerwhile;
                default:
                    System.out.println("Unknown command");
                    break;
            }
            System.out.println();
        }
        input.close();
    }

}

File: 1/recursive.c
/*
    Author: Kalle Elmdahl 21/09/07 (updated 21/09/09)
    The code is used for reversing strings.
    It takes a command-line input and prints it out in reverse using the "printCharactersReverse" function.
    The code is based on an assignment from the KTH-course ID1020
*/

#include <stdio.h>

/*
    Uses the command-line and reverses the input after the enter key is pressed
*/
void printCharactersReverse();

/*
    Used for testing the "printCharactersReverse" function
    The function makes it possible to repeatedly reverse strings without restarting the program
*/
static void test();

/*
    Program startup
*/
int main() {
    test();
    return 0;
}

void printCharactersReverse() {
    int character = getchar();
    if(character == '\n') return;
    printCharactersReverse();
    printf("%c", character);
}

static void test() {
    printCharactersReverse();
    printf("\n\n Again? [y/n]: ");
    int c = getchar();
    if(c == 'y') {
        getchar();
        test();
    }
}